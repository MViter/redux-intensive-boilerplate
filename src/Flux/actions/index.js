export const changePage = (page) => ({
    type:    'CHANGE_PAGE',
    payload: page
});
