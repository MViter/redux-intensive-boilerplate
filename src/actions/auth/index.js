// Types
import types from './types';

export default {
    login: (user) => ({
        type:    types.LOGIN,
        payload: user
    }),
    loginSuccess: () => ({
        type: types.LOGIN_SUCCESS
    }),
    loginFail: (message) => ({
        type:    types.LOGIN_FAIL,
        payload: message
    }),
    signup: (user) => ({
        type:    types.SIGNUP,
        payload: user
    }),
    signupSuccess: () => ({
        type: types.SIGNUP_SUCCESS
    }),
    signupFail: (message) => ({
        type:    types.SIGNUP_FAIL,
        payload: message
    }),
    logout: () => ({
        type: types.LOGOUT
    }),
    logoutSuccess: () => ({
        type: types.LOGOUT_SUCCESS
    }),
    logoutFail: (message) => ({
        type:    types.LOGOUT_FAIL,
        payload: message
    })
};
