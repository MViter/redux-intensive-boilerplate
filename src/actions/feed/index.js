import types from './types';

export default{
    fetchMovies: (type) => ({
        type:    types.FETCH_MOVIES,
        payload: type
    }),
    fetchMoviesSuccess: (movies) => ({
        type:    types.FETCH_MOVIES_SUCCESS,
        payload: movies
    }),
    fetchMoviesFail: (error) => ({
        type:    types.FETCH_MOVIES_FAIL,
        payload: error
    }),

    fetchGenres: () => ({
        type: types.FETCH_GENRES
    }),
    fetchGenresSuccess: (genres) => ({
        type:    types.FETCH_GENRES_SUCCESS,
        payload: genres
    }),
    fetchGenresFail: (error) => ({
        type:    types.FETCH_GENRES_FAIL,
        payload: error
    })
};
