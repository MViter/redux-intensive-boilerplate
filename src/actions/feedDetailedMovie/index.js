import types from './types';

export default{
    fetchDetailedMovie: (movieId) => ({
        type:    types.FETCH_DETAILED_MOVIE,
        payload: movieId
    }),
    fetchDetailedMovieSuccess: (movie) => ({
        type:    types.FETCH_DETAILED_MOVIE_SUCCESS,
        payload: movie
    }),
    fetchDetailedMovieFail: (error) => ({
        type:    types.FETCH_DETAILED_MOVIE_FAIL,
        payload: error
    }),
    clearDetailedMovie: () => ({
        type: types.CLEAR_DETAILED_MOVIE
    })
};
