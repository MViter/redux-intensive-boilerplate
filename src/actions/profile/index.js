// Types
import types from './types';

export default {
    fillProfile: (user) => ({
        type:    types.FILL_PROFILE,
        payload: user
    }),
    updateProfile: (user) => ({
        type:    types.UPDATE_PROFILE,
        payload: user
    }),
    updateProfileSuccess: (user) => ({
        type:    types.UPDATE_PROFILE_SUCCESS,
        payload: user
    }),
    updateProfileFail: (message) => ({
        type:    types.UPDATE_PROFILE_FAIL,
        payload: message
    }),
    updateAvatar: (avatar) => ({
        type:    types.UPDATE_AVATAR,
        payload: avatar
    }),
    updateAvatarSuccess: (url) => ({
        type:    types.UPDATE_AVATAR_SUCCESS,
        payload: url
    }),
    updateAvatarFail: (message) => ({
        type:    types.UPDATE_AVATAR_FAIL,
        payload: message
    })
};
