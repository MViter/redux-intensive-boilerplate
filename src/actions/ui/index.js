// Core
import types from './types';

export default {
    initialize: () => ({
        type: types.INITIALIZE
    }),
    startFetchingAuth: () => ({
        type: types.START_FETCHING_AUTH
    }),
    stopFetchingAuth: () => ({
        type: types.STOP_FETCHING_AUTH
    }),
    startFetchingFeed: () => ({
        type: types.START_FETCHING_FEED
    }),
    stopFetchingFeed: () => ({
        type: types.STOP_FETCHING_FEED
    }),
    startFetchingProfile: () => ({
        type: types.START_FETCHING_PROFILE
    }),
    stopFetchingProfile: () => ({
        type: types.STOP_FETCHING_PROFILE
    }),
    startFetchingAvatar: () => ({
        type: types.START_FETCHING_AVATAR
    }),
    stopFetchingAvatar: () => ({
        type: types.STOP_FETCHING_AVATAR
    }),
    startProfileEditing: () => ({
        type: types.START_PROFILE_EDITING
    }),
    stopProfileEditing: () => ({
        type: types.STOP_PROFILE_EDITING
    }),
    startPasswordEditing: () => ({
        type: types.START_PASSWORD_EDITING
    }),
    stopPasswordEditing: () => ({
        type: types.STOP_PASSWORD_EDITING
    })
};
