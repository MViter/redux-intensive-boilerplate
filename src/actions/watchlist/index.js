import types from './types';

export default{

    fetchWatchlist: () => ({
        type: types.FETCH_WATCHLIST
    }),
    fetchWatchlistSuccess: (movieIDs) => ({
        type:    types.FETCH_WATCHLIST_SUCCESS,
        payload: movieIDs
    }),
    fetchWatchlistFail: (message) => ({
        type:    types.FETCH_WATCHLIST_FAIL,
        payload: message
    }),

    addToWatchlist: (movieToAddId) => ({
        type:    types.ADD_TO_WATCHLIST,
        payload: movieToAddId
    }),
    addToWatchlistSuccess: (movies) => ({
        type:    types.ADD_TO_WATCHLIST_SUCCESS,
        payload: movies
    }),
    addToWatchlistFail: (message) => ({
        type:    types.ADD_TO_WATCHLIST_FAIL,
        payload: message
    }),

    deleteFromWatchlist: (movie) => ({
        type:    types.DELETE_FROM_WATCHLIST,
        payload: movie
    }),
    deleteFromWatchlistSuccess: (movies) => ({
        type:    types.DELETE_FROM_WATCHLIST_SUCCESS,
        payload: movies
    }),
    deleteFromWatchlistFail: (message) => ({
        type:    types.DELETE_FROM_WATCHLIST_FAIL,
        payload: message
    })
};
