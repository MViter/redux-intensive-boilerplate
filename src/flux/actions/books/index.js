export const changePage = (page) => {
    return {
        type:    'CHANGE_PAGE',
        payload: page
    };
};
