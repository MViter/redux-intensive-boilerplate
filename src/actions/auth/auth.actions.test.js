// Core
import actions from './';

const user = {
    id:        '',
    groupId:   '',
    token:     '',
    created:   0,
    avatar:    '',
    firstName: '',
    lastName:  '',
    email:     ''
};
const message = 'message';

describe('auth actions:', () => {
    test(`'login' action creator should produce a corresponding action`, () => {
        expect(actions.login(user)).toMatchSnapshot();
    });
    test(`'loginSuccess' action creator should produce a corresponding action`, () => {
        expect(actions.loginSuccess()).toMatchSnapshot();
    });
    test(`'loginFail' action creator should produce a corresponding action`, () => {
        expect(actions.loginFail(message)).toMatchSnapshot();
    });
    test(`'signup' action creator should produce a corresponding action`, () => {
        expect(actions.signup(user)).toMatchSnapshot();
    });
    test(`'signupSuccess' action creator should produce a corresponding action`, () => {
        expect(actions.signupSuccess()).toMatchSnapshot();
    });
    test(`'signupFail' action creator should produce a corresponding action`, () => {
        expect(actions.signupFail(message)).toMatchSnapshot();
    });
    test(`'logout' action creator should produce a corresponding action`, () => {
        expect(actions.logout()).toMatchSnapshot();
    });
    test(`'logoutSuccess' action creator should produce a corresponding action`, () => {
        expect(actions.logoutSuccess()).toMatchSnapshot();
    });
    test(`'logoutFail' action creator should produce a corresponding action`, () => {
        expect(actions.logoutFail(message)).toMatchSnapshot();
    });
});
