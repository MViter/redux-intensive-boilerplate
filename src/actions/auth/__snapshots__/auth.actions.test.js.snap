// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth actions: 'login' action creator should produce a corresponding action 1`] = `
Object {
  "payload": Object {
    "avatar": "",
    "created": 0,
    "email": "",
    "firstName": "",
    "groupId": "",
    "id": "",
    "lastName": "",
    "token": "",
  },
  "type": "LOGIN",
}
`;

exports[`auth actions: 'loginFail' action creator should produce a corresponding action 1`] = `
Object {
  "payload": "message",
  "type": "LOGIN_FAIL",
}
`;

exports[`auth actions: 'loginSuccess' action creator should produce a corresponding action 1`] = `
Object {
  "type": "LOGIN_SUCCESS",
}
`;

exports[`auth actions: 'logout' action creator should produce a corresponding action 1`] = `
Object {
  "type": "LOGOUT",
}
`;

exports[`auth actions: 'logoutFail' action creator should produce a corresponding action 1`] = `
Object {
  "payload": "message",
  "type": "LOGOUT_FAIL",
}
`;

exports[`auth actions: 'logoutSuccess' action creator should produce a corresponding action 1`] = `
Object {
  "type": "LOGOUT_SUCCESS",
}
`;

exports[`auth actions: 'signup' action creator should produce a corresponding action 1`] = `
Object {
  "payload": Object {
    "avatar": "",
    "created": 0,
    "email": "",
    "firstName": "",
    "groupId": "",
    "id": "",
    "lastName": "",
    "token": "",
  },
  "type": "SIGNUP",
}
`;

exports[`auth actions: 'signupFail' action creator should produce a corresponding action 1`] = `
Object {
  "payload": "message",
  "type": "SIGNUP_FAIL",
}
`;

exports[`auth actions: 'signupSuccess' action creator should produce a corresponding action 1`] = `
Object {
  "type": "SIGNUP_SUCCESS",
}
`;
